src/jit.rs

```diff
@@ -453,14 +455,11 @@ fn declare_variable(
     int: types::Type,
     builder: &mut FunctionBuilder,
     variables: &mut HashMap<String, Variable>,
-    index: &mut usize,
     name: &str,
 ) -> Variable {
-    let var = Variable::new(*index);
+    let var = builder.declare_var(int);
     if !variables.contains_key(name) {
         variables.insert(name.into(), var);
-        builder.declare_var(var, int);
-        *index += 1;
     }
     var
```

Comment:
```suggestion
    variables.entry(name).or_insert_with(|| builder.declare_var(int))
```

Otherwise two references to the same variable name would result in different variables.

Reply:
Fixed. Thanks.